plugins/kubernetes-backend/src/service/KubernetesFetcher.ts

```diff
@@ -216,21 +216,17 @@ export class KubernetesClientBasedFetcher implements KubernetesFetcher {
     let requestInit: RequestInit;
     const authProvider =
       clusterDetails.authMetadata[ANNOTATION_KUBERNETES_AUTH_PROVIDER];
-    if (
-      authProvider === 'serviceAccount' &&
-      !clusterDetails.authMetadata.serviceAccountToken &&
-      fs.pathExistsSync(Config.SERVICEACCOUNT_CA_PATH)
-    ) {
+
+    if (this.isServiceAccountAuthentication(authProvider, clusterDetails)) {
       [url, requestInit] = this.fetchArgsInCluster(credential);
     } else if (
-      credential.type === 'bearer token' ||
-      authProvider === 'localKubectlProxy'
+      this.isTokenOrClientCertificateAuthentication(authProvider, credential)
```

Comment:
Unfortunately this isn't quite the right name for this method, since it also returns true if there's no token or client cert but the auth provider is `localKubectlProxy`. I'm kinda tempted to reverse this helper and name it something like:

```suggestion
      !this.isCredentialMissing(authProvider, credential)
```